src/components/Dialog/Dialog.js

```diff
@@ -91,12 +96,55 @@ const defaultProps = {
   title: undefined,
   position: null,
   onDragStop: () => {},
+  onClickedOutside: () => {},
 };
 
 /**
  * This component creates a Dialog window.
  */
 class Dialog extends Component {
+  constructor(props) {
+    super(props);
+    this.ref = React.createRef();
+    // Bind the function to be able to removeEventListener.
+    this.handleClickOutside = this.handleClickOutside.bind(this);
+  }
+
+  componentDidMount() {
+    const { onClickedOutside, isOpen } = this.props;
+    if (isOpen && onClickedOutside) {
+      document.addEventListener('mousedown', this.handleClickOutside);
+    }
+  }
+
+  componentDidUpdate(prevProps) {
+    const { onClickedOutside, isOpen } = this.props;
+    if (onClickedOutside && prevProps.isOpen !== isOpen) {
+      if (isOpen) {
+        document.addEventListener('mousedown', this.handleClickOutside);
+      } else {
+        document.removeEventListener(
```

Comment:
remove  also the listener on unmount . Create a function activate() and deactivate() to avoid duplicationg code.